packages/react-spinbutton/src/components/SpinButton/useSpinButton.tsx

```diff
@@ -0,0 +1,316 @@
+        internalState.current.spinTime += internalState.current.spinDelay;
+        internalState.current.spinDelay = lerp(
+          DEFAULT_SPIN_DELAY_MS,
+          MIN_SPIN_DELAY_MS,
+          internalState.current.spinTime / MAX_SPIN_TIME_MS,
+        );
+        stepValue(e, direction);
+      }, internalState.current.spinDelay);
+    }
+  };
+
+  const handleIncrementMouseDown = (e: React.MouseEvent<HTMLButtonElement>) => {
+    internalState.current.spinState = 'up';
+    stepValue(e, 'up');
+  };
+
+  const handleDecrementMouseDown = (e: React.MouseEvent<HTMLButtonElement>) => {
+    internalState.current.spinState = 'down';
+    stepValue(e, 'down');
+  };
+
+  const handleStepMouseUpOrLeave = (e: React.MouseEvent<HTMLButtonElement>) => {
+    clearStepTimeout();
+    internalState.current.spinState = 'rest';
+    internalState.current.spinDelay = DEFAULT_SPIN_DELAY_MS;
+    internalState.current.spinTime = 0;
+  };
+
+  const handleBlur = (e: React.FocusEvent<HTMLInputElement>) => {
+    commit(e, currentValue, textValue);
+  };
+
+  const handleKeyDown = (e: React.KeyboardEvent<HTMLInputElement>) => {
+    if (e.key === Keys.ArrowUp) {
+      stepValue(e, 'up');
+      setSpinState('up');
+    } else if (e.key === Keys.ArrowDown) {
+      stepValue(e, 'down');
+      setSpinState('down');
+    } else if (e.key === Keys.PageUp) {
+      e.preventDefault();
+      stepValue(e, 'upPage');
+      setSpinState('up');
+    } else if (e.key === Keys.PageDown) {
+      e.preventDefault();
+      stepValue(e, 'downPage');
+      setSpinState('down');
+    } else if (!e.shiftKey && e.key === Keys.Home && min !== undefined) {
+      commit(e, min);
+      setSpinState('down');
+    } else if (!e.shiftKey && e.key === Keys.End && max !== undefined) {
+      commit(e, max);
+      setSpinState('up');
+    } else if (e.key === Keys.Escape) {
+      if (internalState.current.previousTextValue) {
+        setTextValue(internalState.current.previousTextValue);
+        internalState.current.previousTextValue = undefined;
+      }
+      setSpinState('rest');
+    } else {
+      setSpinState('rest');
+    }
+  };
+
+  const handleKeyUp = (e: React.KeyboardEvent<HTMLInputElement>) => {
+    setSpinState('rest');
+  };
+
+  const commit = (e: SpinButtonChangeEvent, newValue?: number, newDisplayValue?: string) => {
+    const valueChanged = newValue !== undefined && currentValue !== newValue;
+    const displayValueChanged = newDisplayValue !== undefined;
+
+    let roundedValue;
+    if (valueChanged) {
+      roundedValue = precisionRound(newValue!, precision);
+      setCurrentValue(roundedValue);
+      internalState.current.value = roundedValue;
+      setAtBound(getBound(roundedValue, min, max));
+    }
+
+    if (valueChanged || displayValueChanged) {
+      onChange?.(e, { value: roundedValue, displayValue: newDisplayValue });
+    }
+  };
+  state.root.onWheel = useMergedEventCallbacks(state.root.onWheel, handleWheel);
```

Comment:
was it decided to keep the mousewheel behavior in? I agree with what you said before that it doesn't seem wonderful

Reply:
definitely :)